text/0000-refactor-sbom.md

```diff
@@ -0,0 +1,260 @@
+
+or more concretely:
+
+### Example 1 - layer-scoped SBOM file
+
+The `exporter` will turn `/layers/sbom/launch/my-buildpack/launch-dep/sbom.syft.json` into configuration:
+
+```toml
+[[attestations]]
+type = "application/vnd.cyclonedx+json"
+uri = "/layers/sbom/launch/my-buildpack/launch-dep/bom.syft.json"
+
+    [[attestations.annotations]]
+    name = "io.buildpacks.buildpack.id"
+    value = "my-buildpack"
+    
+    [[attestations.annotations]]
+    name = "io.buildpacks.sbom.scope"
+    value = "layer=sha256:$DIGEST" # or maybe layer name? somewhere?
+```
+
+And the `signer` will turn this into an in-toto statement with `predicateType` `https://cyclonedx.org/bom`:
+
+TODO
+
+It's worth noting, many buildpacks-produced SBOM files will be "layer-scoped" meaning that they don't describe the whole image, just a layer within it.
+`cosign` has a [deprecated](https://github.com/sigstore/cosign/blob/main/specs/SBOM_SPEC.md#scopes) method of denoting the scope of an SBOM that is mirrored here.
+However there are probably newer recommended ways of doing this, perhaps within the predicate - this requires further research (TODO).
+
+### Example 2 - image-scoped SBOM file
+
+The `exporter` will turn `/layers/sbom/launch/my-buildpack/launch.sbom.cdx.json` into configuration:
+
+```toml
+[[attestations]]
+type = "application/vnd.cyclonedx+json"
+uri = "/layers/sbom/launch/my-buildpack/launch.sbom.cdx.json"
+
+    [[attestations.annotations]]
+    name = "io.buildpacks.buildpack.id"
+    value = "my-buildpack"
+    
+    [[attestations.annotations]]
+    name = "io.buildpacks.sbom.scope"
+    value = "all"
+```
+
+And the `signer` will turn this into an in-toto statement with `predicateType` `https://cyclonedx.org/bom`:
+
+TODO
+
+### Example 3 - platform-provided SBOM file
+
+Platforms could inject TOML to describe the run image, e.g.:
+
+```toml
+[[attestations]]
+type = "application/vnd.cyclonedx+json"
+uri = "/platform/some-file.cdx.json"
+    
+    [[attestations.annotations]]
+    name = "io.buildpacks.sbom.scope"
+    value = "run" # maybe?
+```
+
+Potentially, the lifecycle could translate the special value `"run"` into a list of layer digests such as: `layer=sha256:$DIGEST-1,layer=sha256:$DIGEST-2`
+to save work for the platform to produce a more meaningful scope.
+
+## Rebase
+
+During `rebase`, the lifecycle should verify attestations for the old application image, and remove any attestations where the scopes are no longer valid
+when signing the new image digest
+(e.g., if the scope `layer=sha256:$DIGEST` references a layer in the old run image that is no longer present, the attestation will be removed).
+
+It should optionally accept configuration for new attestations to describe the new run image (or whatever the platform wants to add).
+
+## Next builds
+
+During `analyze`, the lifecyle should verify attestations for the previous application image, and download the SBOM data as files in `<layers>/sbom/launch/`
```

Comment:
If there is an SBOM in the registry for the run image, we could do interesting things, like provide that to buildpacks (somehow, somewhere) so they could know more about what's available. But that should probably be out of scope for now.